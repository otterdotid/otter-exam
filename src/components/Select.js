import React from 'react';
import PropTypes from 'prop-types';

const Select = ({ data, label, ...props }) => {
  return (
    <React.Fragment>
      <label>{label}</label>
      <select {...props} defaultValue='-'>
        <option defaultValue='-' disabled>
          -
        </option>
        {data ? (
          data.map((list) => (
            <option key={list.id} value={list.id}>
              {list.name}
            </option>
          ))
        ) : (
          <option value='null'>No data available</option>
        )}
      </select>
    </React.Fragment>
  );
};

Select.propTypes = {
  data: PropTypes.array,
  label: PropTypes.string,
};

export default Select;
