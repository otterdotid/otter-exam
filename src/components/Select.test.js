import React from 'react';
import { render } from '@testing-library/react';

import Select from './Select';

describe('<Select />', () => {
  it('renders without crash', () => {
    const wrapper = render(<Select />);
    expect(wrapper.asFragment()).toMatchSnapshot();
  });
});
