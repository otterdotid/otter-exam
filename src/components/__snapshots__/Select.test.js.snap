// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Select /> renders without crash 1`] = `
<DocumentFragment>
  <label />
  <select>
    <option
      value="null"
    >
      No data available
    </option>
  </select>
</DocumentFragment>
`;
